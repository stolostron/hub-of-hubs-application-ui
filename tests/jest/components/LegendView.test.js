// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
"use strict";

import React from "react";
import renderer from "react-test-renderer";

import LegendView from "../../../src-web/components/Topology/viewer/LegendView";

class MockViewContainer {
  getBoundingClientRect() {
    return new Map();
  }
}

describe("LegendView no nodes", () => {
  const returnEmptyArr = jest.fn();
  returnEmptyArr.mockReturnValue([]);
  const viewContainer = jest.fn();
  viewContainer.mockReturnValue(new MockViewContainer());
  const mockData = {
    locale: "en-US",
    handleClose: jest.fn(),
    getLayoutNodes: returnEmptyArr,
    getViewContainer: viewContainer
  };
  it("renders as expected", () => {
    const component = renderer.create(
      <LegendView
        locale={mockData.locale}
        onClose={mockData.handleClose}
        getLayoutNodes={mockData.getLayoutNodes}
        getViewContainer={mockData.getViewContainer}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

const mockLaidoutNodes = {
  laidoutNodes: [
    {
      id: "application--mortgage-app",
      uid: "application--mortgage-app",
      name: "mortgage-app",
      type: "application",
      namespace: "default"
    },
    {
      id: "member--subscription--default--mortgage-app-subscription",
      uid: "member--subscription--default--mortgage-app-subscription",
      name: "mortgage-app-subscription",
      type: "subscription",
      namespace: "default"
    },
    {
      id: "member--rules--default--mortgage-app-placement--0",
      uid: "member--rules--default--mortgage-app-placement--0",
      name: "mortgage-app-placement",
      type: "placements",
      namespace: "default"
    },
    {
      id: "member--clusters--localcluster",
      uid: "member--clusters--localcluster",
      name: "localcluster",
      type: "cluster",
      namespace: ""
    },
    {
      id:
        "member--deployable--member--clusters--localcluster--default--mortgage-app-deployable",
      uid:
        "member--deployable--member--clusters--localcluster--default--mortgage-app-deployable",
      name: "mortgage-app-deployable",
      type: "deployable",
      namespace: "default"
    },
    {
      id:
        "member--member--deployable--member--clusters--localcluster--default--mortgage-app-deployable--deployment--mortgage-app-deploy",
      uid:
        "member--member--deployable--member--clusters--localcluster--default--mortgage-app-deployable--deployment--mortgage-app-deploy",
      name: "mortgage-app-deploy",
      type: "deployment",
      namespace: ""
    },
    {
      id:
        "member--pod--member--deployable--member--clusters--localcluster--default--mortgage-app-deployable--mortgage-app-deploy",
      uid:
        "member--pod--member--deployable--member--clusters--localcluster--default--mortgage-app-deployable--mortgage-app-deploy",
      name: "mortgage-app-deploy",
      type: "deployment",
      namespace: ""
    },
    {
      id:
        "member--deployable--member--clusters--localcluster--default--mortgage-app-svc",
      uid:
        "member--deployable--member--clusters--localcluster--default--mortgage-app-svc",
      name: "mortgage-app-svc",
      type: "deployable",
      namespace: "default"
    },
    {
      id:
        "member--member--deployable--member--clusters--localcluster--default--mortgage-app-svc--service--mortgage-app-svc",
      uid:
        "member--member--deployable--member--clusters--localcluster--default--mortgage-app-svc--service--mortgage-app-svc",
      name: "mortgage-app-svc",
      type: "service",
      namespace: ""
    },
    {
      id: "deploymentconfig",
      uid: "deploymentconfig",
      name: "deploymentconfig",
      type: "deploymentconfig",
      namespace: ""
    },
    {
      id: "replicationcontroller",
      uid: "replicationcontroller",
      name: "replicationcontroller",
      type: "replicationcontroller",
      namespace: ""
    },
    {
      id: "securitycontextconstraints",
      uid: "securitycontextconstraints",
      name: "securitycontextconstraints",
      type: "securitycontextconstraints",
      namespace: ""
    },
    {
      id: "persistentvolumeclaim",
      uid: "persistentvolumeclaim",
      name: "persistentvolumeclaim",
      type: "persistentvolumeclaim",
      namespace: ""
    }
  ]
};

const mockLayoutNodes = jest.fn();
mockLayoutNodes.mockReturnValue(mockLaidoutNodes.laidoutNodes);
const mockData = {
  locale: "en-US",
  handleClose: jest.fn()
};

class MockViewContainer2 {
  getBoundingClientRect() {
    return { height: 667 };
  }
}

const viewContainer2 = jest.fn();
viewContainer2.mockReturnValue(new MockViewContainer2());

describe("LegendView with nodes", () => {
  it("render as expected", () => {
    const component = renderer.create(
      <LegendView
        locale={mockData.locale}
        onClose={mockData.handleClose}
        getLayoutNodes={mockLayoutNodes}
        getViewContainer={viewContainer2}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
