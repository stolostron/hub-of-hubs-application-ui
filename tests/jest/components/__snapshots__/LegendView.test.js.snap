// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendView no nodes renders as expected 1`] = `
<section
  className="topologyDetails"
>
  <div>
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      className="closeIcon"
      description="Close topology legend view"
      fill="currentColor"
      height="1em"
      onClick={[MockFunction]}
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 352 512"
      width="1em"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
      />
    </svg>
    <hr
      style={
        Object {
          "marginBottom": "20px",
          "visibility": "hidden",
        }
      }
    />
  </div>
  <div
    className="legend-view-container"
    style={
      Object {
        "height": "100%",
        "overflow": "hidden",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "bottom": 0,
          "height": "calc(100vh - 300px)",
          "left": 0,
          "marginBottom": 0,
          "marginRight": 0,
          "overflow": "scroll",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <div
        className="legendHeader"
      >
        <div>
          <div
            className="titleText"
          >
            How to read topology
          </div>
          <div
            className="bodyText"
          >
            The topology provides a visual representation of all the applications and resources within a project, their build status, and the components and services associated with them.
          </div>
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <svg>
              <use
                className="label-icon"
                href="#diagramShapes_legend"
              />
            </svg>
          </div>
        </div>
      </div>
      <hr />
      <div
        className="legendBody"
      >
        <div>
          <div
            className="titleText"
          >
            Status icon legend
          </div>
          <div
            className="titleNoteText"
          >
            Note: Resources that you do not have permission to view display a status of "Not deployed".
          </div>
          <div
            className="bodyText"
          >
            <div>
              <svg
                className="statusSvg"
                fill="#3E8635"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_success"
                />
              </svg>
            </div>
            <div>
              All resources in this group have deployed on the target clusters, although their status might not be successful.
            </div>
          </div>
          <div
            className="bodyText"
          >
            <div>
              <svg
                className="statusSvg"
                fill="#878D96"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_pending"
                />
              </svg>
            </div>
            <div>
              None of the resources in this group deployed.
            </div>
          </div>
          <div
            className="bodyText"
          >
            <div>
              <svg
                className="statusSvg"
                fill="#F0AB00"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_warning"
                />
              </svg>
            </div>
            <div>
              Some resources in this group did not deploy. Other resources deployed successfully.
            </div>
          </div>
          <div
            className="bodyText"
          >
            <div>
              <svg
                className="statusSvg"
                fill="#C9190B"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_failure"
                />
              </svg>
            </div>
            <div>
              Some resources in this group are in error state.
            </div>
          </div>
          <div
            className="bodyText"
          >
            For more details and logs, click on the nodes to open the properties view.
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "height": 6,
          "left": 2,
          "position": "absolute",
          "right": 2,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "height": "100%",
            "position": "relative",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "position": "absolute",
          "right": 2,
          "top": 2,
          "width": 6,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "position": "relative",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</section>
`;

exports[`LegendView with nodes render as expected 1`] = `
<section
  className="topologyDetails"
>
  <div>
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      className="closeIcon"
      description="Close topology legend view"
      fill="currentColor"
      height="1em"
      onClick={[MockFunction]}
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 352 512"
      width="1em"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
      />
    </svg>
    <hr
      style={
        Object {
          "marginBottom": "20px",
          "visibility": "hidden",
        }
      }
    />
  </div>
  <div
    className="legend-view-container"
    style={
      Object {
        "height": "100%",
        "overflow": "hidden",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "bottom": 0,
          "height": "calc(100vh - 300px)",
          "left": 0,
          "marginBottom": 0,
          "marginRight": 0,
          "overflow": "scroll",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <div
        className="legendHeader"
      >
        <div>
          <div
            className="titleText"
          >
            How to read topology
          </div>
          <div
            className="bodyText"
          >
            The topology provides a visual representation of all the applications and resources within a project, their build status, and the components and services associated with them.
          </div>
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <svg>
              <use
                className="label-icon"
                href="#diagramShapes_legend"
              />
            </svg>
          </div>
        </div>
      </div>
      <hr />
      <div
        className="legendBody"
      >
        <div>
          <div
            className="titleText"
          >
            Status icon legend
          </div>
          <div
            className="titleNoteText"
          >
            Note: Resources that you do not have permission to view display a status of "Not deployed".
          </div>
          <div
            className="bodyText"
          >
            <div>
              <svg
                className="statusSvg"
                fill="#3E8635"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_success"
                />
              </svg>
            </div>
            <div>
              All resources in this group have deployed on the target clusters, although their status might not be successful.
            </div>
          </div>
          <div
            className="bodyText"
          >
            <div>
              <svg
                className="statusSvg"
                fill="#878D96"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_pending"
                />
              </svg>
            </div>
            <div>
              None of the resources in this group deployed.
            </div>
          </div>
          <div
            className="bodyText"
          >
            <div>
              <svg
                className="statusSvg"
                fill="#F0AB00"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_warning"
                />
              </svg>
            </div>
            <div>
              Some resources in this group did not deploy. Other resources deployed successfully.
            </div>
          </div>
          <div
            className="bodyText"
          >
            <div>
              <svg
                className="statusSvg"
                fill="#C9190B"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_failure"
                />
              </svg>
            </div>
            <div>
              Some resources in this group are in error state.
            </div>
          </div>
          <div
            className="bodyText"
          >
            For more details and logs, click on the nodes to open the properties view.
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "height": 6,
          "left": 2,
          "position": "absolute",
          "right": 2,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "height": "100%",
            "position": "relative",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "borderRadius": 3,
          "bottom": 2,
          "display": "none",
          "position": "absolute",
          "right": 2,
          "top": 2,
          "width": 6,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.2)",
            "borderRadius": "inherit",
            "cursor": "pointer",
            "display": "block",
            "position": "relative",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</section>
`;
